R/render.r

```diff
@@ -370,3 +376,94 @@ footer_pkgdown <- function(data) {
     data$pkgdown$version, "."
   )
 }
+
+data_deps <- function(pkg, depth) {
+
+  # theme variables from configuration
+  bs_version <- pkg$bs_version
+  bootswatch_theme <- pkg$meta[["template"]]$params$bootswatch %||% NULL
+
+  check_bootswatch_theme(bootswatch_theme, bs_version, pkg)
+
+  bs_theme <- do.call(
+    bslib::bs_theme,
+    c(
+      list(
+        version = bs_version,
+        bootswatch = bootswatch_theme
+        ),
+      pkg$meta[["template"]]$bslib
+    )
+  )
+  deps <- bslib::bs_theme_dependencies(bs_theme)
+  # Add other dependencies - TODO: more of those?
+  # Even font awesome had a too old version in R Markdown (no ORCID)
+
+  # Dependencies files end up at the website root in a deps folder
+  deps <- lapply(
+    deps,
+    htmltools::copyDependencyToDir,
+    file.path(pkg$dst_path, "deps")
+  )
+
+  # Function needed for indicating where that deps folder is compared to here
+  transform_path <- function(x) {
+
+    x <- gsub(pkg$dst_path, "", x)
+
+    if (depth == 0) {
+      return(sub("/", "", x))
+    }
+
+    paste0(
+      paste0(rep("..", depth), collapse = "/"), # as many levels up as depth
+      x
+    )
+
+  }
+
+  # Tags ready to be added in heads
+  htmltools::renderDependencies(
+    deps,
+    srcType = "file",
+    hrefFilter = transform_path
+  )
+}
+
+logo_path <- function(pkg, depth) {
+  if (!has_logo(pkg)) {
+    return(NULL)
+  }
+   path <- "package-logo.png"
+
+    if (depth == 0) {
+      return(path)
+    }
+
+    paste0(
+      paste0(rep("..", depth), collapse = "/"), # as many levels up as depth
+      "/",
+      path
+    )
+
+}
+
+check_bootswatch_theme <- function(bootswatch_theme, bs_version, pkg) {
+  if (is.null(bootswatch_theme)) {
+    return(invisible())
+  }
+
+  if (!bootswatch_theme %in% bslib::bootswatch_themes(bs_version)) {
```

Comment:
Needs early return

Reply:
I do not understand this comment: there's an `abort()` right after this?